### ISSUE TITLE
MSE: display/refresh lags

### ISSUE THREAD

**Issue by jlbruce** [USGS-WiM/SiGLDMS#202]
I had just copied in about 30 sites from a spreadsheet, and as I was scrolling to get back to the beginning, the display froze mid-scroll for a while. Not horribly long, but long enough that I was patient for a bit, and then wondered if it had barfed and I needed to reload. I'm not sure if it's really an issue since it did recover eventually, but I had the console open, and I noticed this message that might be helpful?

![display_lag](https://cloud.githubusercontent.com/assets/16122136/19895558/341138bc-a01f-11e6-8b46-d4a5100bae6c.png)

I'm also seeing a substantial lag after large saves - it goes through the progress bar just fine - but then it sits and spins for a while after the bar goes away (sometimes long enough that I go do something else for a minute and come back). Again, not sure anything can be done, it's just unfortunate since we've made this nice graphic that shows them how long they have to wait, and then they have to wait again once it's done.

**Comment by troddick** [comment]
When I changed the functionality from 'once your done saving, go back to the siteList' to 'once your done saving, stay here', I had to do a refresh() on the page so that the user sees the page from a fresh load. Otherwise, if they clicked save and they were scrolled over on the page to parameters or something, the page would just show from where they left off. Doing the refresh also populates the model properly in case they want to do more editing/saving. I can maybe throw up a modal or message right before it refreshes saying that the page must refresh between saves?
